### PR TITLE
Episode click action when episode in episode list is clicked

### PR DESCRIPTION
**Pull request by eliezerben**
When cur episode is clicked, scroll to top.
When any episode in episode list is clicked, autoplay.

### CODE REVIEW

File: lib/screens/episode.dart [hunk 1]
```diff
@@ -90,11 +90,9 @@ class _EpisodeScreenState extends ConsumerState<EpisodeScreen> with AutoRouteAwa
   @override
   void didUpdateWidget(old) {
     super.didUpdateWidget(old);
+    scrollCompleter = wrapInCompleter(scrollController.animateTo(0, duration: const Duration(milliseconds: 600), curve: Curves.easeOutExpo));
```
Comment (andreasgan) [inline lib/screens/episode.dart:93]: you removed setState, why?

File: lib/screens/episode.dart [hunk 2]
```diff
@@ -108,7 +106,7 @@ class _EpisodeScreenState extends ConsumerState<EpisodeScreen> with AutoRouteAwa
       var episode = await episodeFuture;
       if (!mounted || episode == null) return;
 
-      playEpisode(playerId: player!.playerId, autoplay: false, episode: episode, playbackPositionMs: event.playbackPositionMs);
+      playEpisode(playerId: player!.playerId, autoplay: widget.autoplay, episode: episode, playbackPositionMs: event.playbackPositionMs);
```
Comment (andreasgan) [inline lib/screens/episode.dart:109]: undo this, this is for when you stop casting, it's supposed to be false

File: lib/screens/episode.dart [hunk 3]
```diff
@@ -209,7 +207,7 @@ class _EpisodeScreenState extends ConsumerState<EpisodeScreen> with AutoRouteAwa
 
     setState(() {
       playerSetupCompleter = wrapInCompleter(
-          playEpisode(playerId: player.playerId, episode: episode, autoplay: true, playbackPositionMs: startPositionSeconds * 1000)
+          playEpisode(playerId: player.playerId, episode: episode, autoplay: widget.autoplay, playbackPositionMs: startPositionSeconds * 1000)
```
Comment (andreasgan) [inline lib/screens/episode.dart:210]: Probably a reason this was hardcoded to `true`. Don't remember now.

**Review by andreasgan — CHANGES_REQUESTED** [review]
See comments.

File: lib/screens/episode.dart (hunk 1 above)
Comment (eliezerben) [reply lib/screens/episode.dart:93]: According to the documentation, setState is not required inside didUpdateWidget.

> The framework always calls build after calling didUpdateWidget, which means any calls to setState in didUpdateWidget are redundant.

https://api.flutter.dev/flutter/widgets/State/didUpdateWidget.html

**Review by andreasgan — APPROVED** [review]

**Pull request merged**